### PR TITLE
Vehicle update2 Part 1a: Water motion update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This function is sort of the pathfinding along with `sub_427FC9` which hasn't been implemented.

### CODE REVIEW

File: src/OpenLoco/Map/TileManager.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ namespace OpenLoco::Map::TileManager
         tile_coord_t tileX = x / 32;
         tile_coord_t tileY = y / 32;
 
-        size_t index = ((y << 9) | x) >> 5;
+        size_t index = (tileY << 9) | tileX;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Map/TileManager.cpp:29]: Why was the `>> 5` dropped here?

Comment (duncanspumpkin) [reply src/OpenLoco/Map/TileManager.cpp:29]: So I've switched it to tile coords as they are already computed just before this. Tile coords are 32x smaller (>>5). The reason i made the change was i wanted to mask off the additional bits of a full coord.

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 2]
```diff
@@ -553,14 +556,208 @@ namespace OpenLoco::Vehicles
...
+                veh2->sprite_yaw++;
+            }
+            veh2->sprite_yaw &= 0x3F;
+        }
+
+        Vehicle1* veh1 = vehicleUpdate_1;
+        auto distX = (veh2->currentSpeed >> 13) * factorXY503B6A[veh2->sprite_yaw * 2];
+        auto distY = (veh2->currentSpeed >> 13) * factorXY503B6A[veh2->sprite_yaw * 2 + 1];
+
+        veh1->var_4E += distX;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:743]: this calc is wrong. needs to do it as one big calc

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
I tested this using [this big save file]((https://github.com/OpenLoco/OpenLoco/files/6056945/Let.s.Play.Chris.Sawyer.s.Locomotion.Randomland.by.matsku84.-.2088.SV5.zip)). Left the game running for a few months, and compared the autosaves to autosaves for the same months on #751.

Unfortunately, the saves for all months diverge quite a lot. Since no randomness is involved in this PR, there must be a change compared to vanilla. For comparison, see the report in #747.

The first month (near-instant save):
```
Hex Compare
Produced: 28/02/21 3:56:55 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_15-32-48.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2d/autosave_2021-02-28_15-34-12.SV5
2727468 same byte(s)
993 left orphan byte(s)
980 right orphan byte(s)
3453 difference byte(s)
```

The second month:
```
Hex Compare
Produced: 28/02/21 3:57:17 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_15-32-59.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2d/autosave_2021-02-28_15-34-24.SV5
2749596 same byte(s)
5494 left orphan byte(s)
4388 right orphan byte(s)
12555 difference byte(s)
```

The third month:
```
Hex Compare
Produced: 28/02/21 3:57:33 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_15-33-10.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2d/autosave_2021-02-28_15-34-35.SV5
2739886 same byte(s)
8152 left orphan byte(s)
11264 right orphan byte(s)
24262 difference byte(s)
```

**Comment by AaronVanGeffen** [comment]
Unforunately, 14be2a67f3fe5ecbf66708a525204e360ca5168f does not fix the issue.

First month:
```
Hex Compare
Produced: 28/02/21 7:43:32 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_19-40-14.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_19-42-06.SV5
2460371 same byte(s)
990 left orphan byte(s)
1003 right orphan byte(s)
3441 difference byte(s)
```

Second month:
```
Hex Compare
Produced: 28/02/21 7:43:57 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_19-40-26.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_19-42-17.SV5
2482505 same byte(s)
4462 left orphan byte(s)
5572 right orphan byte(s)
12479 difference byte(s)
```

Third month:
```
Hex Compare
Produced: 28/02/21 7:44:19 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_19-40-38.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/vehicle_update2a/autosave_2021-02-28_19-42-29.SV5
2472839 same byte(s)
11316 left orphan byte(s)
8211 right orphan byte(s)
24144 difference byte(s)
```

**Review by AaronVanGeffen — APPROVED** [review]
I tested this using [this big save file]((https://github.com/OpenLoco/OpenLoco/files/6056945/Let.s.Play.Chris.Sawyer.s.Locomotion.Randomland.by.matsku84.-.2088.SV5.zip)). Left the game running for a few months, and collected three monthly autosaves on this branch. I did the same thing for master. After, I then compared the autosaves for the same months.

The first month (near-instant save):
```
Hex Compare
Produced: 28/02/21 10:23:08 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/duncan/autosave_2021-02-28_22-21-17.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_22-20-27.SV5
2464797 same byte(s)
5 difference byte(s)
```

The second month:
```
Hex Compare
Produced: 28/02/21 10:23:20 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/duncan/autosave_2021-02-28_22-21-28.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_22-20-38.SV5
2499439 same byte(s)
1 left orphan byte(s)
7 difference byte(s)
```

The third month:
```
Hex Compare
Produced: 28/02/21 10:23:31 pm 

Left file: /home/aaron/.config/OpenLoco/save/autosave/duncan/autosave_2021-02-28_22-21-40.SV5
Right file: /home/aaron/.config/OpenLoco/save/autosave/master/autosave_2021-02-28_22-20-50.SV5
2508295 same byte(s)
4 difference byte(s)
```

**Pull request merged**